crates/wasi-component-loader/src/extension/instance.rs

```diff
@@ -33,22 +33,21 @@ impl ExtensionInstance {
         directive: wit::FieldDefinitionDirective<'_>,
         inputs: InputList,
     ) -> crate::Result<FieldOutput> {
+        // Futures may be canceled, so we pro-actively mark the instance as poisoned until proven
+        // otherwise.
+        self.poisoned = true;
+
```

Comment:
Would be great to have some helper to do this and not DRY so much. But it might be annoying to write, so...